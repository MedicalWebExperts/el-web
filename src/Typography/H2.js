import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';

const propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.oneOf(['primary', 'secondary', 'default']),
};

const defaultProps = {
  color: 'default',
};

const styles = theme => ({
  headline: {
    fontSize: theme.fonts.baseSize * 2,
    color: props => theme.colors[props.color],
  },
});

const H2 = ({ classes, text }) => (
  <h2 className={classes.headline}>{text}</h2>
);

H2.propTypes = propTypes;
H2.defaultProps = defaultProps;

export default injectSheet(styles)(H2);
