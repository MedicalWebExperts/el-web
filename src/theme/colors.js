export default {
  default: '#000000',
  primary: '#0000ff',
  secondary: '#00b33c',
};
