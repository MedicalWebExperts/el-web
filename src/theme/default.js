export default {
  colors: {
    primary: '#0000ff',
    secondary: '#00b33c',
    default: '#000000',
  },
  font: {
    primary: 'Georgia, serif',
    secondary: 'sans-serif',
  },
};
