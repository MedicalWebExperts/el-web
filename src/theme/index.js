import { withTheme } from 'theming';
import ThemeProvider from './provider';
import DefaultTheme from './default';

module.exports = {
  ThemeProvider,
  withTheme,
  DefaultTheme,
};
