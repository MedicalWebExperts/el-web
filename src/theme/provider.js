import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as ThemeProviderBase } from 'theming';
import DefaultTheme from './default';

const propTypes = {
  children: PropTypes.element.isRequired,
  theme: PropTypes.oneOfType([PropTypes.shape({})]),
};

const defaultProps = {
  theme: {},
};

const ThemeProvider = ({ children, theme }) => {
  // check a util funcion for deep clone/merge
  const mergedTheme = Object.keys(theme).length === 0 ? DefaultTheme : theme;

  return (
    <ThemeProviderBase theme={mergedTheme}>
      {children}
    </ThemeProviderBase>
  );
};

ThemeProvider.propTypes = propTypes;
ThemeProvider.defaultProps = defaultProps;

export default ThemeProvider;
