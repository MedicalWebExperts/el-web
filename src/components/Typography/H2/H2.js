import React from 'react';
import PropTypes from 'prop-types';
import { withTheme } from '../../../theme';

const propTypes = {
  text: PropTypes.string.isRequired,
  secondary: PropTypes.bool,
  color: PropTypes.oneOf(['primary', 'secondary', 'default']),
  theme: PropTypes.shape({}).isRequired,
};

const defaultProps = {
  color: 'default',
  secondary: false,
};

const H2 = ({
  text,
  theme,
  color,
  secondary,
}) => (
  <h2>
    <style jsx>{`
      h2 {
        color: ${theme.colors[color]}
        font-family: ${secondary ? theme.font.secondary : theme.font.primary}
      }
    `}
    </style>
    {text}
  </h2>
);

H2.propTypes = propTypes;
H2.defaultProps = defaultProps;

export default withTheme(H2);
