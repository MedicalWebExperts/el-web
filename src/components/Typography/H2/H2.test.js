import React from 'react';
import renderer from 'react-test-renderer';

import H2 from './H2';
import { ThemeProvider } from '../../../theme';

describe('Component H2 Snapshot', () => {
  it('without arguments, take the default values', () => {
    const tree = renderer
      .create(<ThemeProvider><H2 text="Heading 2" /></ThemeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('primary color of the theme', () => {
    const tree = renderer
      .create(<ThemeProvider><H2 text="Heading 2" color="primary" /></ThemeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('secondary font-family of the theme', () => {
    const tree = renderer
      .create(<ThemeProvider><H2 text="Heading 2" secondary /></ThemeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
