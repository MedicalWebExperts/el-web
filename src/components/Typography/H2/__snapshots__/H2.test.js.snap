// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component H2 Snapshot primary color of the theme 1`] = `
<h2
  className="jsx-1074868122"
>
  Heading 2
</h2>
`;

exports[`Component H2 Snapshot secondary font-family of the theme 1`] = `
<h2
  className="jsx-1123045924"
>
  Heading 2
</h2>
`;

exports[`Component H2 Snapshot without arguments, take the default values 1`] = `
<h2
  className="jsx-3769017946"
>
  Heading 2
</h2>
`;
